code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -24,12 +24,35 @@
 	var/isbroken = 0
 	light_range = 5
 	light_color = LIGHT_COLOR_RED
-	var/obj/item/wepon = null
+	var/last_check = 0
+
+/obj/structure/cult/pylon/New()
+	..()
+	processing_objects.Add(src)
+
+/obj/structure/cult/pylon/Destroy()
+	..()
+	processing_objects.Remove(src)
+
+/obj/structure/cult/pylon/process()
+	if(!isbroken && world.time > last_check + 3 SECONDS)
+		last_check = world.time
+		for(var/mob/living/simple_animal/construct/C in view(src, 3))
+			if(C.health < C.maxHealth)
+				if(prob(15))
+					src.visible_message("<span class='sinister'>\the [src] mends some of \the <EM>[C]'s</EM> wounds.</span>")
+				make_tracker_effects(get_turf(src), C)
+				C.health = min(C.maxHealth, C.health + 3) //Not quite as good as artificers
+
 
 /obj/structure/cult/pylon/attack_hand(mob/M as mob)
 	attackpylon(M, 5)
 
 /obj/structure/cult/pylon/attack_animal(mob/living/simple_animal/user as mob)
+	if(istype(user, /mob/living/simple_animal/construct/builder))
+		if(isbroken && prob(20))
```

Comment:
Am I deliberately only meant to have a small chance to break the pylon, breaking it otherwise? That might be frustrating, better put a message for the fail case if so.